src/utility/NTPUtils.cpp

```diff
@@ -0,0 +1,72 @@
+#include "NTPUtils.h"
+#include "Arduino.h"
+
+// could be a constexpr in C++14
+static time_t cvt_TIME(char const *time) {
+    char s_month[5];
+    int month, day, year;
+    struct tm t = {0};
+    static const char month_names[] = "JanFebMarAprMayJunJulAugSepOctNovDec";
+
+    sscanf(time, "%s %d %d", s_month, &day, &year);
+
+    month = (strstr(month_names, s_month)-month_names)/3;
+
+    t.tm_mon = month;
+    t.tm_mday = day;
+    t.tm_year = year - 1900;
+    t.tm_isdst = -1;
+
+    return mktime(&t);
+}
+
+
+NTPUtils::NTPUtils(UDP& Udp) : Udp(Udp) {}
+
+bool NTPUtils::isTimeValid(unsigned long time) {
+    Serial.println("Compile time: " + String(cvt_TIME(__DATE__)));
+    return (time > cvt_TIME(__DATE__));
+}
+
+void NTPUtils::sendNTPpacket(uint8_t* packetBuffer) {
+  memset(packetBuffer, 0, NTP_PACKET_SIZE);
+  packetBuffer[0] = 0b11100011;
```

Comment:
Lots of **magic numbers** ;) Could we at least provide a link to some documentation on the web where the format of a NTP packet is explained?

Reply:
Agree again, the code is from https://www.arduino.cc/en/Tutorial/UdpNTPClient (to avoid including yet another library for such a small functionality). Could adding the link to the complete tutorial be enough in your opinion?

Reply:
Suits me ;) I did not know the code used originated from an existing tutorial. As there is already a complete tutorial available a link over there should do the trick just fine.

Reply:
@lxrobotics @facchinm 
where should the link be placed?
right at the top of `NTPutils.h` or `.cpp`? `.h` makes more sense to me but it's not where the _magic_ numbers are